### PR TITLE
PD-1477 / 25.04 / Pd 1477 update nextcloud tutorial

### PR DESCRIPTION
**Pull request by micjohnson777**
This PR updates the instructions in the Nextcloud app tutorial to reflect the current process with the new Docker Compose app deployment.

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

**Comment by micjohnson777** [comment]
Tutorial updated with postres storage volume ACL settings. Requesting a review from Starvros to make sure nothing is missed or incorrect. The instructions in this tutorial deploy a working container for Nextcloud.

File: static/includes/apps/AppInstallWizardACLConfiguration.md
Comment (stavros-k) [inline static/includes/apps/AppInstallWizardACLConfiguration.md:15]: The `473` user is only used on minio. But its not the rule for stable apps.

The rule is something like that:

All apps that can be run as any non-root user, default to `568`.

Apps that can run only as a specific non-root user, are run as this specific user. Which there is no option in the UI to change that.

Apps that need root privileges, run as root. Which there is no option in the UI to change that.

Postgres and MariaDB on all apps runs as 999:999, regardless of the Run As setting (This only affects the main container).

File: static/includes/apps/AppInstallWizardACLConfiguration.md
Comment (stavros-k) [inline static/includes/apps/AppInstallWizardACLConfiguration.md:17]: If user creates TrueNAS user with uid 3005, but the app runs as 568. This wont work.

The entry must match the run-as UID/GID.
(On apps that there is no fields in the UI to change, the UID can be seen on the Run As Context in the screen right before installing or right after installing)

File: static/includes/apps/AppsInstallWizardDefaultPorts.md
Comment (stavros-k) [inline static/includes/apps/AppsInstallWizardDefaultPorts.md:4]: There is no longer a limitation on ports below 9000.
User can effectively assign 1-65535 (as long as its not already used).

0-1024 might require the application to have elevated privileges.

File: static/includes/apps/AppInstallWizardACLConfiguration.md
Comment (micjohnson777) [reply static/includes/apps/AppInstallWizardACLConfiguration.md:15]: I'll get these updates made! Thanks for the feedback!

File: static/includes/apps/AppInstallWizardACLConfiguration.md
Comment (micjohnson777) [reply static/includes/apps/AppInstallWizardACLConfiguration.md:17]: Changes made in the next commit.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:102]: On my system **Language** is just a text entry field, no dropdown with language options.

https://ocrmypdf.readthedocs.io/en/latest/languages.html#installing-additional-language-packs
Says that typically English is supported by default but you may have to install other language packs? Might want to check in with Stavros on how this field works and then link to either that page or https://github.com/tesseract-ocr/tesseract/blob/main/doc/tesseract.1.asc#languages for more on language choices.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:104]: Can be a FQDN:Port or IP:Port

Tooltip says 
>cloud.domain.com:30001
cloud.domain.com (if you use port 443 externally)
192.168.1.100:9001 (replace ip and port with your own)

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:122]:
```suggestion
Enter an open port number or accept the default port, **30027**, in **WebUI Port**.
This port must match the one used in **Host** above.
```

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:111]: "Accept the remaining defaults" is a little confusing, because it feels like that's it, I'm done setting up the app, but then we're really just eliding a few fields before talking about others that do need attention.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:124]:
```suggestion
If you configured a certificate, select it in **Certificate ID**. A certificate is required if you want to select an external port other than the default **30027**.
```

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:136]: Image shows **Force Flag** selected and it is not mentioned in the text. Same for the next dataset.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:149]: Maybe just

```suggestion
Accept the defaults in **Resources Configuration**, and select the GPU option if applicable.
```

They might have a GPU installed but not want it allocated to Nextcloud.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:166]:
```suggestion
Nextcloud configuration settings include setting up credentials, ACT packages (previously referred to as the commands), the host IP and port, data directory path, upload limits, execution times, memory limits and cache memory consumption, adding a cron job with schedule, and adding additional environment variables.
```

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:241]:
```suggestion
Earlier deployments of the Nextcloud app use five datasets, the parent dataset for the application (**nextcloud**) and the four child datasets:
```

**Review by DjP-iX — CHANGES_REQUESTED** [review]
I worked through the entire process. I wasn't able to finally confirm the install succeeds due to the docker rate limit, so I can do that in the morning, but here are my "as a reader" comments/suggestions for now.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:70]: I didn't want to include this in the before you begin section but there is no reason not to.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:32]: I structured this snippet this way because certificates are required for some apps and optional for others. If I insert the (optional) in the bullet point I cannot use the same snippet in all app tutorials. I've modified the sentence to include (if required) at the end it.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:114]: changed this in the next commit

**Comment by micjohnson777** [comment]
Suggested changes made in the last commit. Stavros plans to review this later today/tomorrow so wait for his additional feedback before merging.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:39]: html for app data
data for **user** data
postgres_data for **database** data

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:75]: Is this intended to be here? Its also on the top. But still feels weird to be here.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:106]: Hmm UI does not seem to display my description (tooltip). I'll open an issue for that.
But here is the description, feel free to copy what needed to adapt docs.

```yaml
          description: |
            The tesseract languages to install for OCRmypdf.</br>
            See a list of language codes here https://tesseract-ocr.github.io/tessdoc/Data-Files-in-different-versions.html.</br>
            Typing a wrong language code will block the container from starting.</br>
            Only takes effect if ocrmypdf is selected above.
            ```

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:143]: 1001 is not correct here.
Nextcloud runs as root, but the subprocess of nextcloud runs as `33/www-data` 
So either no ACLs are needed, or use 33/www-data 
(nextcloud will _attempt_ to fix the permissions itself on the first install in either case. we dont control that)

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:148]: not 1001

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:192]: Dont we have this section above?
Also there is a screenshot of nextcloud at login page, but then we go again over installation settings?

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:298]: Not touched in this PR, but we might need to clarify that this was about Dragonfish only.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:328]:
```suggestion
Within the Postgres container, the user and group 999 map to **postgres**.
```

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:327]: There isn't a postgres backup dataset anymore

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:75]: Yes, this is the instruction for creating a new user. The similar name of the snippet file might be throwing you off. ;-)

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:106]: Added the information provided in the Nextcloud Configuration Settings section.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:144]: About user ID **1001** in the Nextcloud tutorial, I found this UID in the questions.yml file and using it does not prevent deploying the app. Should I say to enter this or **0** for root user? We don't enter the string the way you provided it 33/www-data, so I guess it would be just **33** entered as the ACE entry UID.  I can also provide the instructions to add this user from the Edit ACL screen like it is for the postgres_data dataset.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:302]: I'll remove this section. the 24.04 docs has the old tutorial with this content so it is still available in that branch.

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (stavros-k) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md:144]: Its possible 1001 was a typo. But its no longer there on docker apps!

33 is better yes, tho nextcloud will still attempt to fix perms on install.

Comment (micjohnson777) [reply content/TruenasApps/StableApps/InstallNextCloudMedia.md]: changed 1001 to 33

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:109]: I think the # has the potential to add confusion rather than help

```suggestion
Enter either the fully qualified domain name or the IP address and port for your TrueNAS system in **Host** as *##.###.###.##:port* or *my.domain.com:port*.
```

File: content/TruenasApps/StableApps/InstallNextCloudMedia.md
Comment (DjP-iX) [inline content/TruenasApps/StableApps/InstallNextCloudMedia.md:137]:
```suggestion
{{< hint type=info >}}
Do not select **DEPRECATED: Old Storage Structure** if you are deploying Nextcloud for the first time as this slows down the installation and is not necessary.
If you are upgrading where your Nextcloud deployment in TrueNAS was a 1.x.x release, select this option.
{{< /hint >}}
```

**Review by DjP-iX — APPROVED** [review]

**Pull request merged**
